Jenkins/Jenkinsfile-template-linux-multibranch

```diff
@@ -17,30 +17,35 @@ pipeline {
   agent any
 
   environment {
-    // The Jenkins Credentials ID (as a secret text credential) for your GitHub PAT. The PAT needs `repo` scope.
-    GITHUB_CREDS = credentials('c066da44-44dd-4188-b000-e6e63d42dffb')
+    // The Jenkins Credentials ID (as a secret text credential) for your GitHub PAT.
+    GITHUB_CREDS = credentials('github-pat-as-secret-text')
     // The repo default branch name
     DEFAULT_BRANCH = 'main'
     // The type of ref that will be checked out for a job initiated by a GitHub PR. 'merge' for the default PR merge commit, 'head' for the head of the branch being merged.
-    GITHUB_PR_REF_TYPE = 'merge'
+    GITHUB_PR_REF_TYPE = 'head'
     PR_REF = getPRRef()
     // The name of the GitHub repository to run the analysis on
-    GITHUB_REPO = 'od-ctcampbell/demo-jenkins-1'
+    GITHUB_REPO = 'example/java-app'
+    MAVEN_ARGS = '-batch-mode --no-transfer-progress'
+    MAVEN_OPTS = '-Dmaven.test.skip=true'
+    // See https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/customizing-your-advanced-setup-for-code-scanning#changing-the-languages-that-are-analyzed
+    // for supported languages.
+    // For Java and Kotlin use 'java', for JavaScript and TypeScript use 'javascript', for C and C++ use 'cpp'.
     CODEQL_LANGUAGE = 'java'
     // See https://docs.github.com/en/code-security/codeql-cli/codeql-cli-manual/database-create#--build-modemode for build mode options
     CODEQL_BUILD_MODE = 'manual'
-    CODEQL_BUILD_COMMAND = './mvnw clean package -B'
+    CODEQL_BUILD_COMMAND = './mvnw clean package'
     CODEQL_QUERY_SUITE = "codeql/${CODEQL_LANGUAGE}-queries:codeql-suites/${CODEQL_LANGUAGE}-security-extended.qls"
-    CODEQL_BUNDLE_URL = 'https://github.com/github/codeql-action/releases/latest/download/codeql-bundle-linux64.tar.gz'
-    CODEQL_BIN_DIR = '/codeql'
-    PATH = "${CODEQL_BIN_DIR}:$PATH"
+    // CODEQL_BUNDLE_URL = 'https://github.com/github/codeql-action/releases/latest/download/codeql-bundle-linux64.tar.gz'
+    CODEQL_DIR = '/tools/codeql'
```

Comment:
Do you want to add a comment around "CodeQL is recommended to store in a tool cache so you don't download it every time"?